linkcheck/checker/urlbase.py

```diff
@@ -632,8 +632,10 @@ def get_content(self):
         if self.text is None:
             self.get_raw_content()
             self.soup = htmlsoup.make_soup(self.data)
-            self.text = self.data.decode(self.soup.original_encoding)
-            self.encoding = self.soup.original_encoding
+            # Sometimes soup.original_encoding is None!  Better mangled text
+            # than an internal crash, eh?
+            self.encoding = self.soup.original_encoding or 'ISO-8859-1'
```

Comment:
i have a hunch as to why you picked latin1 here, but could you document it? (my guess is: latin-1 can decode any byte string without errors, as opposed to utf8)

Reply:
That is correct.

This segues to me opening a can of worms: what if BeautifulSoup gives us an encoding, but for some reason that encoding is wrong and .decode() raises a UnicodeDecodeError?  Should we use .decode(encoding, 'replace')?  Or do we want to wait for bug reports and investiagate them?

Reply:
>  Or do we want to wait for bug reports and investigate them?

I'm tempted to pick this option because it means no more work for me right now ;)